bodo/pandas/physical/read_iceberg.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <Python.h>
+#include <arrow/compute/api.h>
+#include <arrow/python/pyarrow.h>
+#include <memory>
+#include <utility>
+#include "../../io/iceberg_parquet_reader.h"
+#include "duckdb/planner/bound_result_modifier.hpp"
+#include "duckdb/planner/table_filter.hpp"
+#include "operator.h"
+
+/// @brief Physical node for reading Parquet files in pipelines.
+class PhysicalReadIceberg : public PhysicalSource {
+   private:
+    const std::shared_ptr<arrow::Schema> arrow_schema;
+    const std::unique_ptr<IcebergParquetReader> internal_reader;
+    static std::vector<std::string> _create_out_column_names(
+        const std::vector<int> &selected_columns,
+        const std::shared_ptr<arrow::Schema> schema);
+    static std::unique_ptr<IcebergParquetReader> _create_internal_reader();
```

Comment:
Minor: using underscore in the beginning of C++ function looks quite strange (coming from Python habits).